ui-admin/src/portal/siteContent/AddPageModal.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState } from 'react'
+import Modal from 'react-bootstrap/Modal'
+import { HtmlPage, PortalEnvironment } from '@juniper/ui-core'
+import Api from 'api/api'
+import { useConfig } from 'providers/ConfigProvider'
+
+/** renders a modal that adds a new page to the site */
+const AddPageModal = ({ portalEnv, portalShortcode, insertNewPage, show, setShow }: {
+  portalEnv: PortalEnvironment, portalShortcode: string, insertNewPage: (page: HtmlPage) => void,
+  show: boolean, setShow:  React.Dispatch<React.SetStateAction<boolean>>
+}) => {
+  const zoneConfig = useConfig()
+
+  const [pageTitle, setPageTitle] = useState('')
+  const [pagePath, setPagePath] = useState('')
+
+  const addPage = async () => {
+    insertNewPage({
+      title: pageTitle,
+      path: pagePath,
+      sections: []
+    })
+    setShow(false)
+  }
+
+  const clearFields = () => {
+    setPageTitle('')
+    setPagePath('')
+  }
+
+  const portalUrl = Api.getParticipantLink(portalEnv.portalEnvironmentConfig, zoneConfig.participantUiHostname,
+    portalShortcode, portalEnv.environmentName)
+
+  return <Modal show={show}
```

Comment:
There are probably other options that we'll want to surface in this modal in the future, like allowing the user to set the new page as the landing page or not add it to the navbar. But for now this should be enough?